### ISSUE TITLE
Can't use classes defined in jars

### ISSUE THREAD

**Issue by doug-wade** [albertlatacz/java-repl#91]
I'm trying to use GSON/Guava in my repl from command line, but I can't seem to get them on the classpath.  I've tried using the `:cp` command (per [this issue](https://github.com/albertlatacz/java-repl/issues/50)), and also adding the `-cp` argument to `java`

```
dougwade ~/browser-extensions » java -cp ".:~/bin/*" -jar ~/bin/javarepl.jar
Welcome to JavaREPL version 292 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_60)
Type expression to evaluate, :help for more options or press tab to auto-complete.
java> :cp file:/Users/doug.wade/bin/
Added file:/Users/doug.wade/bin/ to classpath.
java> import com.google.gson.Gson;
ERROR: package com.google.gson does not exist
import com.google.gson.Gson;;
                      ^
java> Gson gson = new Gson();
ERROR: cannot find symbol
  symbol:   class Gson
  location: interface Evaluation
  Gson method$27mujvxti59rkq0sc6ey();
  ^
```

I know that the jars are, in fact, in /Users/doug.wade/bin

```
dougwade ~/browser-extensions » ls /Users/doug.wade/bin
RedshiftJDBC41-1.1.7.1007.jar  gson-2.5.jar  guava-19.0.jar  javarepl.jar
```

I've tried both `-cp` and `:cp` with and without the \* wildcard.  I've also tried importing Joiner from Guava with the same result.  I tried rooting around through the tests for an example of adding a jar to the classpath, but a github search for `cp` yields nothing in `/test`.

I'm sure there's something obvious I'm missing, but I for the life of me can't figure out what I'm doing wrong here.

**Comment by spark85** [comment]
I've got the same problem using the Android studio plugin & from the command line

**Comment by chuckwondo** [comment]
When specifying a directory, it behaves just like you would expect for the usual Java classpath in that a directory on the classpath will work only for exploded classes under that directory, not for jar files. To add jar files to the classpath, specify each one individually. You should not need to specify the `file:` prefix either. I believe the following (or similar) should work in the case you've given:

```
:cp /Users/doug.wade/bin/RedshiftJDBC41-1.1.7.1007.jar
:cp /Users/doug.wade/bin/gson-2.5.jar
:cp /Users/doug.wade/bin/guava-19.0.jar
```

Adding anything to the classpath of the java command that starts javarepl has no effect because javarepl creates a separate class loader for use within the repl, as far as I understand.

**Comment by attheveryend** [comment]
i'm trying to import selenium into the repl via jar as made available from maven, and I've tried every permutation of the classpath command I can think of and still can't get it to import anything at all.

**Comment by chuckwondo** [comment]
@attheveryend Can you provide more details? Where is the jar file on the file system? How are you attempting to import it? Are you getting any error message?

**Comment by attheveryend** [comment]
Using Win7-64bit

So I went to the maven repository webzone to collect all selenium related jar files ( https://mvnrepository.com/artifact/org.seleniumhq.selenium ), and tried to use CP command within the repl runtime to get things into the repl, and then use an import command to gain access to the libraries.  I tried stating the path to my .jars in various ways (using \ , /, and //, etc.).  I tried using the CP function via windows CMD java -cp and that did not give me access to any of the selenium libraries either.  When I say, "does not give me access," I mean that the import command fails with "package does not exist," meaning the repl can't see the jar file.  I found that the cp: command accepts pretty much anything as input and has no verification step.

I did finally get it to work for FireFox with the following steps:

FIRST:  put all your selenium related jar files in the same folder as javarepl.jar
SECOND: start the java repl and paste the following lines of code into it.  

:cp cglib-nodep-3.2.4.jar
:cp client-combined-3.0.0-beta2-nodeps.jar
:cp commons-codec-1.10.jar
:cp commons-exec-1.3.jar
:cp commons-logging-1.2.jar
:cp gson-2.3.1.jar
:cp guava-19.0.jar
:cp hamcrest-core-1.3.jar
:cp hamcrest-library-1.3.jar
:cp httpclient-4.5.2.jar
:cp httpcore-4.4.4.jar
:cp httime-4.5.2.jar
:cp jna-4.1.0.jar
:cp jna-platform-4.1.0.jar
:cp junit-4.12.jar
:cp netty-3.5.7.Final.jar
:cp phantomjsdriver-1.2.1.jar
:cp selenium-chrome-driver-3.0.0.jar
:cp selenium-firefox-driver-3.0.0.jar
:cp selenium-ie-driver-3.0.0.jar
:cp selenium-java-3.0.0.jar
:cp selenium-remote-driver-3.0.0.jar


import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
